VimeoUpload/Upload/Descriptor System/VideoDescriptorFailureTracker.swift

```diff
@@ -79,63 +85,67 @@ import Foundation
 
         super.init()
         
-        self.failedDescriptors = self.load()
+        self.failedDescriptors = self.load(folderName: name, migrator: migrator)
 
         self.addObservers()
     }
     
     // MARK: Setup
     
-    private static func setupArchiver(name: String, archivePrefix: String?, documentsFolderURL: URL) -> KeyedArchiver?
+    private static func setupArchiver(folderName: String, archivePrefix: String?, documentsFolderURL: URL) -> KeyedArchiver?
     {
-        let typeFolderURL = documentsFolderURL.appendingPathComponent(name)
+        let folderURL = documentsFolderURL.appendingPathComponent(folderName)
         
-        if FileManager.default.fileExists(atPath: typeFolderURL.path) == false
+        if FileManager.default.fileExists(atPath: folderURL.path) == false
         {
             do
             {
-                try FileManager.default.createDirectory(at: typeFolderURL, withIntermediateDirectories: true, attributes: nil)
+                try FileManager.default.createDirectory(at: folderURL, withIntermediateDirectories: true, attributes: nil)
             }
             catch
             {
                 return nil
             }
         }
         
-        return KeyedArchiver(basePath: typeFolderURL.path, archivePrefix: archivePrefix)
+        return KeyedArchiver(basePath: folderURL.path, archivePrefix: archivePrefix)
     }
     
-    private func load() -> [String: Descriptor]?
+    private func load(folderName: String, migrator: ArchiveMigrating?) -> [String: Descriptor]
     {
         guard self.shouldLoadArchive == true else
         {
-            return nil
+            return [:]
+        }
+        
+        guard let failedDescriptors = ArchiveDataLoader.loadData(relativeFolderPath: folderName,
+                                                                 archiver: self.archiver,
+                                                                 key: VideoDescriptorFailureTracker.ArchiveKey,
+                                                                 migrator: migrator) as? [String: Descriptor]
+        else
+        {
+            return [:]
```

Comment:
Same here, just curious about the change to move away from returning an optional?

Reply:
Same comment above @jasonhawkins .